src/NuGet.Core/NuGet.Packaging/Signing/Package/SignedPackageArchive.cs

```diff
@@ -95,7 +95,8 @@ public Task<bool> IsZip64Async(CancellationToken token)
                 throw new SignatureException(Strings.SignedPackageUnableToAccessSignature);
             }
 
-            using (var reader = new BinaryReader(ZipReadStream, new UTF8Encoding(), leaveOpen: true))
+            using (var bufferedStream = new ReadOnlyBufferedStream(ZipReadStream, leaveOpen: true))
```

Comment:
Adding the changes to `AddSignatureAsync` shouldn't introduce more risk since that code path only gets executed from nuget.exe. I would vote to add it here, but it's not necessary.

Reply:
You're probably right, but I want to minimize churn for this PR.